test_defopt.py

```diff
@@ -78,7 +82,7 @@ def bad(**kwargs):
             defopt.run(bad)
 
     def test_no_function(self):
-        with self.assertRaisesRegex(ValueError, 'at least one'):
+        with self.assertRaises(TypeError):
             defopt.run()
```

Comment:
This is a good point.  However, I would rather just drop that check:
1. it is rather unlikely that one would ever pass an empty list to `defopt.run`.
2. it should be up to argparse to decide whether a subparsers object with no actual subparsers is an error; if argparse thinks it's fine (and it's just as useless for argparse as it is for defopt) then we shouldn't change that.